### PR TITLE
Fix # 353 - Add abi_compatible_with concept

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: test/unit/meta/abi_compatible_with.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#elif defined(SPY_SIMD_IS_ARM)
+TTS_CASE_TPL( "Check that wide<T,N,x86*> satisfies abi_compatible_with with any ARM ABI"
+            , TTS_NUMERIC_TYPES
+            )
+{
+  using eve::wide;
+  using eve::logical;
+
+  TTS_EXPECT((eve::abi_compatible_with<wide<T>          , eve::arm_64_ , eve::arm_128_> ));
+  TTS_EXPECT((eve::abi_compatible_with<logical<wide<T>> , eve::arm_64_ , eve::arm_128_> ));
```
Comment (DenisYaroshevskiy) [inline test/unit/meta/abi_compatible_with.cpp:58]: Where are the false tests?

File: include/eve/concept/abi.hpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                                        >
+                              >
+              || ...
+              )
+            >
+    {};
+  }
+
+  template<typename Wide, typename... ABI>
+  concept abi_compatible_with = detail::is_abi_compatible<Wide,ABI...>::value;
```
Comment (DenisYaroshevskiy) [inline include/eve/concept/abi.hpp:55]: This name doesn't quite make sence.
Why compatible? This checks that the cardinal is the expected one? That it's not smaller or larger?
`expected` `natural` `default`?

File: test/unit/meta/abi_compatible_with.cpp (hunk 1 above)
Comment (jfalcou) [reply test/unit/meta/abi_compatible_with.cpp:58]: Good point

File: include/eve/concept/abi.hpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#include <eve/traits/as_wide.hpp>
+#include <eve/forward.hpp>
+#include <concepts>
+
+namespace eve
+{
+  namespace detail
+  {
+    template<typename Wide, typename... ABI>
+    struct  is_abi_compatible
```
Comment (DenisYaroshevskiy) [inline include/eve/concept/abi.hpp:23]: This should probably be renamed as well?

File: test/unit/meta/native_simd_for_abi.cpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+#if defined(SPY_SIMD_IS_X86)
+TTS_CASE_TPL( "Check that wide<T,N,x86*> satisfies native_simd_for_abi with any X86 ABI"
+            , TTS_NUMERIC_TYPES
+            )
+{
+  using eve::wide;
+  using eve::logical;
+
+  TTS_EXPECT((eve::native_simd_for_abi<wide<T>          , eve::x86_128_ , eve::x86_256_> ));
+  TTS_EXPECT((eve::native_simd_for_abi<logical<wide<T>> , eve::x86_128_ , eve::x86_256_> ));
```
Comment (DenisYaroshevskiy) [inline test/unit/meta/native_simd_for_abi.cpp:36]: I would like to see also tests for bigger and smaller wide.

File: include/eve/concept/abi.hpp (hunk 3 above)
Comment (jfalcou) [reply include/eve/concept/abi.hpp:23]: Sure

File: test/unit/meta/native_simd_for_abi.cpp (hunk 4 above)
Comment (jfalcou) [reply test/unit/meta/native_simd_for_abi.cpp:36]: Yes, also the test need to move from concept to arch as currently is is only tested once in scalar mode :/

**Review by DenisYaroshevskiy — APPROVED** [review]

**Review by DenisYaroshevskiy — APPROVED** [review]

**Pull request merged**